overlord/state/notices.go

```diff
@@ -107,15 +119,84 @@ func flattenUserID(userID *uint32) (uid uint32, isSet bool) {
 	return *userID, true
 }
 
+// Type represents a group of notices originating from a common source.
+func (n *Notice) Type() NoticeType {
+	return n.noticeType
+}
+
+// Key is a string that differentiates notices of the same type.
+func (n *Notice) Key() string {
+	return n.key
+}
+
+// FirstOccurred is the first time this notice occurs.
+func (n *Notice) FirstOccurred() time.Time {
+	return n.firstOccurred
+}
+
+// LastOccurred is the last time this notice occurred.
+func (n *Notice) LastOccurred() time.Time {
+	return n.lastOccurred
+}
+
+// LastRepeated is the time this notice was last "repeated".
+func (n *Notice) LastRepeated() time.Time {
+	return n.lastRepeated
+}
+
+// Occurrences is the number of times this notice has occurred.
+func (n *Notice) Occurrences() int {
+	return n.occurrences
+}
+
+// LastData is additional data captured from the last occurrence of this notice.
+func (n *Notice) LastData() map[string]string {
+	return n.lastData
+}
+
+// RepeatAfter is how long after this notice was last repeated should we allow
+// it to repeat.
+func (n *Notice) RepeatAfter() time.Duration {
+	return n.repeatAfter
+}
+
+// ExpireAfter is how long since this notice last occurred until we should drop it.
+func (n *Notice) ExpireAfter() time.Duration {
+	return n.expireAfter
+}
+
 // expired reports whether this notice has expired (relative to the given "now").
 func (n *Notice) expired(now time.Time) bool {
 	return n.lastOccurred.Add(n.expireAfter).Before(now)
 }
 
-// jsonNotice exists so we can control how a Notice is marshalled to JSON. It
-// needs to live in this package (rather than the API) because we save state
-// to disk as JSON.
-type jsonNotice struct {
+// GetRepeatCheckValue gets current RepeatCheckData.
+//
+// NOTE: "value" must be JSON unmarshallable.
```

Comment:
The NOTE read a bit strangely as the value is a target here. Maybe the doc comment should be more  like the ones for Change.Get etc:

GetRepeatCheckData unmarshals the previously stored repeat check data (from AddNoticeOptions or RepeatCheck) into value.
    value parameter

Reply:
Updated, Thanks!